src/audio/src/src.c

```diff
@@ -1082,3 +1008,623 @@ UT_STATIC void sys_comp_src_init(void)
 }
 
 DECLARE_MODULE(sys_comp_src_init);
+
+#else
+static inline void src_update_buffer_position(struct input_stream_buffer *input_buffers,
+					      struct output_stream_buffer *output_buffers,
+					      int *n_read, int *n_written)
+{
+	struct audio_stream __sparse_cache *source = input_buffers->data;
+	struct audio_stream __sparse_cache *sink = output_buffers->data;
+
+	input_buffers->consumed += audio_stream_frame_bytes(source) * (*n_read);
+	output_buffers->size += audio_stream_frame_bytes(sink) * (*n_written);
+}
+
+/* Fallback function */
+static void src_fallback(struct processing_module *mod, struct input_stream_buffer *bsource,
+			 struct output_stream_buffer *bsink, int *n_read, int *n_written)
+{
+	*n_read = 0;
+	*n_written = 0;
+	src_update_buffer_position(bsource, bsink, n_read, n_written);
+}
+
+/* Normal 2 stage SRC */
+static void src_2s(struct processing_module *mod, struct input_stream_buffer *bsource,
+		   struct output_stream_buffer *bsink, int *n_read, int *n_written)
+{
```

Comment:
can we try to avoid duplicating these functions and create a common function instead and call it from two wrappers - for IPC3 and IPC4 / module adapter?

Reply:
updated